api/tests/LinkyLink.Integration.Tests/LinkApiTests.cs

```diff
@@ -0,0 +1,212 @@
+using LinkyLink.Models;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Identity.Client;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+
+namespace LinkyLink.Integration.Tests
+{
+    [TestClass]
+    public class LinkApiTests : IntegrationTest, IDisposable
+    {
+        private readonly HttpClient client;
+        private readonly AzureAdB2C b2cConfig;
+
+        public LinkApiTests()
+        {
+            this.b2cConfig = this.Configuration.GetSection("AzureAdB2C").Get<AzureAdB2C>();
+
+            this.ValidateConfigurationItem(() => this.b2cConfig.BaseAddress, nameof(this.b2cConfig.BaseAddress));
+            this.ValidateConfigurationItem(() => this.b2cConfig.Authority, nameof(this.b2cConfig.Authority));
+            this.ValidateConfigurationItem(() => this.b2cConfig.ClientId, nameof(this.b2cConfig.ClientId));
+            this.ValidateConfigurationItem(() => this.b2cConfig.Scope, nameof(this.b2cConfig.Scope));
+            this.ValidateConfigurationItem(() => this.b2cConfig.Username, nameof(this.b2cConfig.Username));
+            this.ValidateConfigurationItem(() => this.b2cConfig.Password, nameof(this.b2cConfig.Password));
+
+            this.client = new HttpClient
+            {
+                BaseAddress = new Uri(this.b2cConfig.BaseAddress)
+            };
+        }
+
+        [TestMethod]
+        public async Task Links_API_CRUD_Operations()

```

Comment:
I wonder if it would be better if these were broken down into separate integration tests per endpoint?

Reply:
The intention was to run them in a particular sequence so we can clean up after ourselves.

Reply:
These integration tests / functional tests are essentially to run a smoke test against the API's to ensure they are working.  More intensive tests with positive/negative test cases should typically be done at the unit test level.